mesonbuild/modules/cmake.py

```diff
@@ -416,5 +542,91 @@ def subproject(self, state, args, kwargs):
     def subproject_options(self, state, args, kwargs) -> CMakeSubprojectOptions:
         return CMakeSubprojectOptions()
 
+    @FeatureNew('generate_export', '0.62.0')
+    @typed_pos_args('cmake.generate_export', varargs=dependencies.InternalDependency, min_varargs=1)
+    @typed_kwargs(
+        'cmake.generate_export',
+        KwargInfo('name', (str, NoneType)),
+        KwargInfo('namespace', (str, NoneType)),
+        KwargInfo('subdirs', ContainerTypeInfo(list, str), default=[], listify=True)
+    )
+    def generate_export(self, state, args: T.Tuple[T.List[dependencies.InternalDependency]], kwargs: 'GenerateExport'):
+        deps = args[0]
+
+        coredata = state.environment.get_coredata()
+
+        namespace = kwargs['namespace']
+        if not namespace:
+            namespace = state.project_name
+
+        targets: T.List[T.Union[build.SharedLibrary, build.StaticLibrary]] = []
+        expected_targets = ''
+        # These options are space-separated lists because they are passed to CMake's target_*()
+        compile_options: T.List[str] = []
+        link_options: T.List[str] = []
+
+        for dep in deps:
+            if len(dep.libraries) != 1 or not isinstance(dep.libraries[0], (build.SharedLibrary, build.StaticLibrary)):
+                raise mesonlib.MesonException('cmake.generate_export only supports dependencies wrapping a single library')
+
+            target = dep.libraries[0]
+            targets.append(target)
+            expected_targets += namespace + '::' + target.name + ' '
+
+            target_compile_options = ''
+            for compile_arg in dep.compile_args:
+                target_compile_options += compile_arg + ' '
+            compile_options.append(target_compile_options.strip())
+
+            target_link_options = ''
+            for link_arg in dep.link_args:
+                target_link_options += link_arg + ' '
+            link_options.append(target_link_options.strip())
+        expected_targets = expected_targets.strip()
+
+        name = kwargs['name']
+        if not name:
+            if len(deps) > 1:
+                raise mesonlib.MesonException('When passing multiple deps to cmake.generate_export, the name kwarg is required')
+            name = deps[0].libraries[0].name
+
+        includedir = coredata.get_option(mesonlib.OptionKey('includedir'))
+        include_dirs = f'"${{_IMPORT_PREFIX}}/{includedir}" '
+        for subdir in kwargs['subdirs']:
+            include_dirs += f'"${{_IMPORT_PREFIX}}/{includedir}/{subdir}" '
+        include_dirs = include_dirs.strip()
+
+        targets_file = TARGETS_INIT.replace('@expected_targets@', expected_targets)
+        targets_file = targets_file.replace('@prefix@', coredata.get_option(mesonlib.OptionKey('prefix')))
```

Comment:
This returns the install prefix with the "correct" path separators depending on the platform (`\` on Windows), but CMake uses `/` everywhere. It is possible to always get the path with unix file separators? Or am I forced to convert `\` to `/` when building on Windows?

Reply:
CMake itself supports unix seperators. Also, windows does since a specific version.

Reply:
Yes, but python/meson converts them to backslashes on Windows, while CMake uses Posix separators everywhere. How can I tell python/meson to unconditionally return Posix separator?

Reply:
From [stackoverflow](https://stackoverflow.com/questions/54671385/convert-windowspath-to-posixpath) use: `pathlib.path(r'foo\bar').as_posix()`

Reply:
Oh ok, simpler than expected :D

Reply:
The pkg-config file generator also uses as_posix() for the same reason.